src/types/redis_set.cc

```diff
@@ -338,19 +338,43 @@ rocksdb::Status Set::Inter(const std::vector<Slice> &keys, std::vector<std::stri
   for (const auto &member : target_members) {
     member_counters[member] = 1;
   }
+
+  bool has_limit = limit != 0;
+  bool limited = false;
   for (size_t i = 1; i < keys.size(); i++) {
     s = Members(keys[i], &target_members);
     if (!s.ok() || target_members.empty()) return s;
     for (const auto &member : target_members) {
-      if (member_counters.find(member) == member_counters.end()) continue;
+      if (member_counters.count(member) == 0) continue;
       member_counters[member]++;
+      if (has_limit && member_counters[member] == keys.size()) {
```

Comment:
Why do we search for the `member` element three times in `member_counters` continuously? It looks so bad to me.

```c++
auto iter = find(member); // find it
use1(iter); // use it
use2(iter);

// **NOT**:
use1(find(member));
use2(find(member));
use3(find(member));
```